src/frontend/komponenter/Fagsak/Vilkårsvurdering/GeneriskVilkår/GeneriskVilkårVurdering.tsx

```diff
@@ -0,0 +1,201 @@
+import React, { useState } from 'react';
+import deepEqual from 'deep-equal';
+import { IFelt, Valideringsstatus } from '../../../../typer/felt';
+import { IVilkårResultat, Resultat, IVilkårConfig } from '../../../../typer/vilkår';
+import { useVilkårsvurdering } from '../../../../context/Vilkårsvurdering/VilkårsvurderingContext';
+import { validerVilkår } from '../../../../context/Vilkårsvurdering/validering';
+import { RadioGruppe, Radio, TextareaControlled, SkjemaGruppe } from 'nav-frontend-skjema';
+import {
+    vilkårResultatFeilmeldingId,
+    vilkårBegrunnelseFeilmeldingId,
+    vilkårFeilmeldingId,
+} from './GeneriskVilkår';
+import FastsettPeriode from './FastsettPeriode/FastsettPeriode';
+import { Knapp } from 'nav-frontend-knapper';
+import { IPerson } from '../../../../typer/person';
+import classNames from 'classnames';
+import UtførKnapp from './UtførKnapp';
+import { Undertekst } from 'nav-frontend-typografi';
+import { periodeToString } from '../../../../typer/periode';
+
+interface IProps {
+    person: IPerson;
+    tillattFjerning: boolean;
+    vilkårFraConfig: IVilkårConfig;
+    vilkårResultat: IFelt<IVilkårResultat>;
+    visFeilmeldinger: boolean;
+}
+
+const GeneriskVilkårVurdering: React.FC<IProps> = ({
+    person,
+    tillattFjerning,
+    vilkårFraConfig,
+    vilkårResultat,
+    visFeilmeldinger,
+}) => {
+    const { fjernVilkår, settVilkårForPeriodeResultat } = useVilkårsvurdering();
+
+    const [ekspandertVilkår, settEkspandertVilkår] = useState(
+        vilkårResultat.verdi.resultat.verdi === Resultat.KANSKJE
+    );
+    const [visFeilmeldingerForEttVilkår, settVisFeilmeldingerForEttVilkår] = useState(false);
+
+    const [redigerbartVilkår, settRedigerbartVilkår] = useState<IFelt<IVilkårResultat>>(
+        vilkårResultat
+    );
+
+    const validerOgSettRedigerbartVilkår = (endretVilkår: IFelt<IVilkårResultat>) => {
+        settRedigerbartVilkår(validerVilkår(endretVilkår));
+    };
+
+    const radioOnChange = (resultat: Resultat) => {
+        validerOgSettRedigerbartVilkår({
+            ...redigerbartVilkår,
+            verdi: {
+                ...redigerbartVilkår.verdi,
+                resultat: {
+                    ...redigerbartVilkår.verdi.resultat,
+                    verdi: resultat,
+                },
+            },
+        });
+    };
+
+    const toggleForm = (visAlert: boolean) => {
+        if (ekspandertVilkår && visAlert && !deepEqual(vilkårResultat, redigerbartVilkår)) {
+            alert('Vurderingen har endringer som ikke er lagret!');
```

Comment:
Shouldn't we allow the user to drop the changes and close the form?

Reply:
As of now you can press "avbryt" which lets the user discard the changes, but "lukk" is not allowed until the changes are "saved".